src/selectors/getCallStackFrames.js

```diff
@@ -39,23 +40,30 @@ function appendSource(sources, frame, selectedSource) {
 export function formatCallStackFrames(
   frames: Frame[],
   sources: SourcesMap,
-  selectedSource: Source
+  selectedSource: Source,
+  frameworkGroupingOn: boolean
 ) {
   if (!frames) {
     return null;
   }
 
-  return frames
+  let formattedFrames = frames
     .filter(frame => getSourceForFrame(sources, frame))
     .map(frame => appendSource(sources, frame, selectedSource))
     .filter(frame => !get(frame, "source.isBlackBoxed"))
     .map(annotateFrame);
+
+  if (frameworkGroupingOn) {
+    formattedFrames = collapseFrames(formattedFrames);
```

Comment:
i think the reason i avoided doing this earlier was that the collapsing messed with the count of frames which was important for expanding and collapsing.

The other reason was that it made toggling `frameworkGroupingOn` a bit different... 

This way works, but I wonder if it is needed. If the main benefit of doing the collapsing in the selector is testing, could we re-think how we're doing the testing?

for instance what is the input to the unit test? if it is just an array of frames which have the library field, we dont need to think about annotateFrame or the selector at all.

Reply:
>i think the reason i avoided doing this earlier was that the collapsing messed with the count of frames which was important for expanding and collapsing.
I suspected this might be the case but couldn't find any scenario where this problem would occur. I'm not 100% confident though so I guess I shouldn't be changing that. 

>This way works, but I wonder if it is needed. If the main benefit of doing the collapsing in the selector is testing, could we re-think how we're doing the testing? for instance what is the input to the unit test? if it is just an array of frames which have the library field, we dont need to think about annotateFrame or the selector at all.
The issue with that is, deciding what library a frame belongs to is at least half of the logic for framework grouping. I think we should still have tests to cover that. So, one test suite to make sure frames are annotated properly, and another to ensure frames are grouped based on the annotation. 

It's less than ideal to test against frame annotation, because that's an implementation detail. That was my thinking for this PR. Then again, maybe I've changed more than I need to for the "good enough" solution :/

Reply:
makes sense